internal/db/repos/project.go

```diff
@@ -58,8 +65,11 @@ func (r *ProjectRepository) List(ctx context.Context, ownerID uint, opts *models
 		return nil, fmt.Errorf("invalid owner_id: %w", err)
 	}
 	var projects []models.Project
-	err := r.db.WithContext(ctx).Where(models.Project{OwnerID: ownerID}).
-		Limit(opts.Limit).Offset(opts.Offset).Find(&projects).Error
+	query := r.db.WithContext(ctx).Where(models.Project{OwnerID: ownerID})
+	if opts != nil {
+		query = query.Limit(opts.Limit).Offset(opts.Offset)
+	}
```

Comment:
It would be safer if instead of not passing limit & offset, if it is empty, use a default value to avoid retrieving the entire table from the DB.

Reply:
since it is out of the scope of this PR and this probably is repeated in some other places in the codebase, it could be done in a followup issue. @coderabbitai can you please create a followup for this?

Reply:
https://github.com/celestiaorg/talis/issues/232